src/foam/nanos/auth/CapabilityAuthService.js

```diff
@@ -136,23 +161,36 @@ foam.CLASS({
           // 1. check if there is a capability matching the name of the permission
           // that is enabled and not deprecated, and granted to the user
           Capability cap = (Capability) capabilityDAO.find(EQ(foam.nanos.crunch.Capability.NAME, permission));
-
-          Predicate capabilityScope = AND(
-            EQ(UserCapabilityJunction.SOURCE_ID, user.getId()),
-            OR(
+          Predicate userPredicate = EQ(UserCapabilityJunction.SOURCE_ID, user.getId());
+          Predicate capabilityScope = OR(
               NOT(HAS(UserCapabilityJunction.EXPIRY)),
               NOT(EQ(UserCapabilityJunction.STATUS, CapabilityJunctionStatus.EXPIRED))
-            )
           );
 
           if ( cap != null && cap.getEnabled() ) {
 
             if ( userCapabilityJunctionDAO.find(
               AND(
+                userPredicate,
                 capabilityScope,
                 EQ(UserCapabilityJunction.TARGET_ID, cap.getId()),
                 EQ(UserCapabilityJunction.STATUS, CapabilityJunctionStatus.GRANTED)
-              )) != null ) result = true;
+              )) != null ) {
+              result = true;
+            } else if ( cap != null && agent != null && agent.getId() != user.getId() ) {
+              // if the agent is in the context and is not the same as user, check 
+              // if the agent has capability
+              userPredicate = EQ(UserCapabilityJunction.SOURCE_ID, agent.getId());
+              key = agent.getId() + permission;
+              if ( userCapabilityJunctionDAO.find(
+                AND(
+                  userPredicate,
+                  capabilityScope,
+                  EQ(UserCapabilityJunction.TARGET_ID, cap.getId()),
+                  EQ(UserCapabilityJunction.STATUS, CapabilityJunctionStatus.GRANTED)
+                )) != null )
+                result = true;
```

Comment:
I think we'll run into unwanted accessibility issues with this logic if I understand things correctly. 

Considering a user's relationship with realUser and its granted capabilities, prerequisites that are required from these capabilities can be satisfied by the user and the realUser capabilities. A realUser could potentially have had capabilities granted to them using another users capabilities and then later allowing them to be used with another user which would be an issue in some scenarios, example being signing officer. If I'm a signing officer (given this is handle by a capability) in one business doesn't entail that I'm a signing officer on another. How are we currently handling this? 

Summarized: This can lead to a short handed capability fulfillment where a capability was granted due to the capability prerequisites being satisfied by another users capability and that capabilities granted to realUser does not potentially mean they have it granted (or should be applied) when acting as another user.

A resolution I thought of is to add an `exclusiveTo` reference property to the UCJ which can reference another user within the system signifying whether they have access to those capabilities while realUser is acting as a user.

Reply:
thanks, added to ticket to look into tmr.
https://nanopay.atlassian.net/browse/NP-1335 
I added an extension of UCJ called AgentCapabilityJunction in another PR, this contains an extra property called businessId (could use a better name). I think it may be similar to what you're suggesting.
That was for another (similar) purpose, but you're right that there should be a check for this in the authservice.

Reply:
pr for this https://github.com/Nauna/foam2/pull/83